jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/EndPoint.java

```diff
@@ -324,5 +326,52 @@ default void write(Callback callback, ByteBuffer... buffers) throws WritePending
      *
      * @param newConnection the connection to upgrade to
      */
-    public void upgrade(Connection newConnection);
+    void upgrade(Connection newConnection);
+
+    /**
+     * An endpoint that may be Secure.
+     */
+    interface Securable extends EndPoint
+    {
+        /**
+         * Get the SslSessionData of a secure end point.
+         * @return A {@link SslSessionData} instance (with possibly null field values) if secure, else {@code null}.
+         */
+        SslSessionData getSslSessionData();
+
+        default boolean isSecure()
+        {
+            return getSslSessionData() != null;
+        }
+    }
+
+    /**
+     * Bundle of SSLSession associated data that may be known from an {@link SSLSession} or determined otherwise in the
+     * absence of an {@link SSLSession}.  All fields may be {@code null}, and typically it is the presence of a non
+     * {@code null} instance that indicates a {@link Connection} and/or request is secure.
+     * @param sslSession The {@link SSLSession} itself, if known, else {@code null}
+     * @param sessionId The {@link SSLSession#getId()} rendered as a hex string, if known, else {@code null}
+     * @param cipherSuite The {@link SSLSession#getCipherSuite()} if known, else {@code null}
+     * @param peerCertificates The {@link SSLSession#getPeerCertificates()}s converted to {@link X509Certificate}, if known
+     *                         else {@code null}.
+     */
+    record SslSessionData(SSLSession sslSession, String sessionId, String cipherSuite, X509Certificate[] peerCertificates)
```

Comment:
I agree about the `record`.  I was going to change to a class, but thought this PR was already big enough.... but I guess that horse had already bolted :(

However, I don't agree about the name.  I know TLS is the current name and SSL is effectively dead.   But everything else is named SSL: the JVM classes, our implementations etc. etc.   So I find it confusing whenever we use TLS.    I think we should stick with SSL until such time as we can do a complete change to TLS everywhere.

Reply:
>  I think we should stick with SSL until such time as we can do a complete change to TLS everywhere.

Then it will never happen, as the JDK classes won't be renamed.

Reply:
So we should keep it aligned with our SslConnection* and SslContext* classes.  We might one day rename them.